pkg/api/server/v1alpha2/results.go

```diff
@@ -285,8 +285,11 @@ func (s *Server) getFilteredPaginatedSortedResults(ctx context.Context, parent s
 
 func getResultByParentName(gdb *gorm.DB, parent, name string) (*db.Result, error) {
 	r := &db.Result{}
-	if err := errors.Wrap(gdb.Where(&db.Result{Parent: parent, Name: name}).First(r).Error); err != nil {
-		return nil, status.Errorf(status.Code(err), "failed to query on database: %v", err)
+	q := gdb.
+		Where(&db.Result{Parent: parent, Name: name}).
+		First(r)
+	if err := errors.Wrap(q.Error); err != nil {
+		return nil, err
```

Comment:
Do we have a means of unit testing this error message?

Reply:
@sayan-biswas bump on this question.

Reply:
This errors are actually returned from the gorm package. In this repo we only have a wrapper. Untill we have a predefined set of error messages for all the RPCs, we can't create unit test to check the message string.